src/test/java/com/capitalone/dashboard/event/TestResultEventListenerTest.java

```diff
@@ -0,0 +1,175 @@
+    @Mock
+    private CollectorItemRepository collectorItemRepository;
+    @Mock
+    private PerformanceRepository performanceRepository;
+    @InjectMocks
+    private TestResultEventListener testResultEventListener;
+    private static final String COLLECTOR_NAME = "PerfTools";
+
+
+    @Test
+    public void test_testResultListener(){
+        TestResult testResult = getTestResult(getTestResultCollectorItem());
+        CollectorItem perfCollItem = getNewPerfCollectorItem(getPerfToolsCollector());
+        setupData();
+        testResultEventListener.onAfterSave(new AfterSaveEvent<>(testResult, null, ""));
+        Performance performance = testResultEventListener.createPerformanceDoc(testResult,
+                getTestCapabilities().iterator().next(), perfCollItem);
+        verify(performanceRepository, never()).save(performance);
+        assertThat(performance.getUrl(), is(testResult.getUrl()));
+        assertThat(performance.getType(), is(PerformanceType.ApplicationPerformance));
+    }
+
+    private CollectorItem getTestResultCollectorItem() {
+        CollectorItem collectorItem = new CollectorItem();
+        collectorItem.setId(ObjectId.get());
+        collectorItem.setCollector(new Collector("Test Collector", CollectorType.Test));
+        collectorItem.setNiceName("Test Collector");
+        collectorItem.getOptions().put("jobName", "TestResultEventListenerTest");
+        return collectorItem;
+    }
+
+    private TestResult getTestResult(CollectorItem collectorItem) {
+        TestResult testResult = new TestResult();
+        testResult.setId(ObjectId.get());
+        testResult.setType(TestSuiteType.Performance);
+        testResult.setCollectorItemId(collectorItem.getId());
+        testResult.setTestCapabilities(getTestCapabilities());
+        return testResult;
+    }
+
+    private Collector getPerfToolsCollector() {
+        Collector collector = new Collector(COLLECTOR_NAME, CollectorType.AppPerformance);
+        collector.setId(ObjectId.get());
+        return collector;
+    }
+
+    private void setupData() {
+        when(collectorItemRepository.findOne(Matchers.any(ObjectId.class))).thenReturn(getTestResultCollectorItem());
+        when(collectorRepository.save(Matchers.any(Collector.class))).thenReturn(getPerfToolsCollector());
+        when(collectorItemRepository.save(Matchers.any(CollectorItem.class))).thenReturn(getNewPerfCollectorItem(getPerfToolsCollector()));
+        when(performanceRepository.save(Matchers.any(Performance.class))).thenReturn(getPerformanceDoc());
+    }
+
+    public CollectorItem getNewPerfCollectorItem(Collector perfToolsCollector) {
+        CollectorItem newPerfCollItem = new CollectorItem();
+        newPerfCollItem.setId(ObjectId.get());
+        newPerfCollItem.setCollector(perfToolsCollector);
+        newPerfCollItem.setCollectorId(perfToolsCollector.getId());
+        return newPerfCollItem;
+    }
+
+    public Collection<TestCapability> getTestCapabilities() {
```

Comment:
this can be more modular. you can move suite, test vase & steps to other methods

Reply:
Thanks for the comment. Its a unit test method with no PMD violations or NPath complexity issues.